AnkiDroid/src/main/java/com/ichi2/anki/noteeditor/Toolbar.java

```diff
@@ -201,19 +205,43 @@ public AppCompatImageButton insertItem(@IdRes int id, Drawable drawable, Runnabl
         // end apply style
 
 
-        this.mToolbar.addView(button, mToolbar.getChildCount());
-        mCustomButtons.add(button);
-        button.setOnClickListener(l -> runnable.run());
-
         // Hack - items are truncated from the scrollview
         View v = findViewById(R.id.editor_toolbar_internal);
 
-        int expectedWidth = getVisibleItemCount() * dpToPixels(48 + 2 * 4); //width + 4dp padding on both sides
+        int expectedWidth = getVisibleItemCount(-1) * dpToPixels(48 + 2 * 4); //width + 4dp padding on both sides
         int width = getScreenWidth();
-        LayoutParams p = new LayoutParams(v.getLayoutParams());
+        LinearLayout.LayoutParams p = new LinearLayout.LayoutParams(v.getLayoutParams());
         p.gravity = Gravity.CENTER_VERTICAL | ((expectedWidth > width) ? Gravity.START : Gravity.CENTER_HORIZONTAL);
         v.setLayoutParams(p);
 
+        if (expectedWidth <= width) {
+            this.mToolbar.addView(button, mToolbar.getChildCount());
+        }
+        else {
```

Comment:
Nit: style guide: https://github.com/ankidroid/Anki-Android/wiki/Code-style

```suggestion
        } else {
```